fix(upgrade): handle output emitters when downgrading a component

crisbeto opened the pull request:
The `DowngradeComponentAdapter` adapter was assuming that all outputs are observables, but they can also be `OutputEmitterRef`.

Fixes #60366.

crisbeto commented on packages/upgrade/src/common/src/downgrade_component_adapter.ts:
```diff
@@ -258,13 +259,16 @@ export class DowngradeComponentAdapter {
+        : (v: any) => getter(this.scope, {'$event': v});
+
+      const subscription =
+        emitter instanceof OutputEmitterRef
```
I went with checking for `OutputEmitterRef`, rather than `EventEmitter` here, because technically any observable can be an `@Output`.

JoostK replied:
Couldn't we just always pass the closure directly object wrapper. Rxjs observables support that so it's subclass `EventEmitter` should too?

crisbeto replied:
Ah now I remember, I think the signature with three arguments is deprecated. The one with a single argument should be fine. I'll switch it over.

crisbeto committed:
fix(upgrade): handle output emitters when downgrading a component

The `DowngradeComponentAdapter` adapter was assuming that all outputs are observables, but they can also be `OutputEmitterRef`.

Fixes #60366.

thePunderWoman reviewed (approved):
LGTM

pkozlowski-opensource commented:
This PR was merged into the repository by commit cd7c170763b9f1f0b4215bf151a431481591ce73.

The changes were merged into the following branches: main, 19.2.x
